sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/client/TokenExchange.java

```diff
@@ -63,30 +65,39 @@ public static synchronized void getDbToken(
             TokenExchangeServiceCallback serviceCallback) {
         AppCenterLog.debug(LOG_TAG, "Getting a resource token from App Center...");
         String url = apiUrl + GET_TOKEN_PATH_FORMAT;
-
-        ServiceCall tokenResponse =
-            httpClient.callAsync(
-                url,
-                METHOD_POST,
-                new HashMap<String, String>() { { put(APP_SECRET_HEADER, appSecret); } },
-                    new HttpClient.CallTemplate() {
-
-                        @Override
-                        public String buildRequestBody() {
-                            return buildAppCenterGetDbTokenBodyPayload(partition);
-                        }
-
-                        @Override
-                        public void onBeforeCalling(URL url, Map<String, String> headers) { }
-                    },
-                    serviceCallback);
+        TokenResult tokenResult = TokenManager.getInstance().getCachedToken(partition);
+        if (tokenResult != null) {
+            serviceCallback.callCosmosDb(tokenResult);
+        } else {
+            ServiceCall tokenResponse =
```

Comment:
I'm not sure this formatting is correct... Make sure you run the formatter.

Reply:
This is the correct formatting... run the format locally... few places changed, but not here..

Reply:
I will commit the formatted file